src/Core/Managed/Shared/Extensibility/Implementation/BroadcastProcessor.cs

```diff
@@ -0,0 +1,148 @@
+﻿namespace Microsoft.ApplicationInsights.Shared.Extensibility.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+    /// <summary>
+    /// An <see cref="ITelemetryProcessor"/> that asynchronously sends the data to multiple child telemetry processors.
+    /// </summary>
+    internal class BroadcastProcessor : ITelemetryProcessor, IDisposable
```

Comment:
Well, technically, you could still implement thread-affinity back-pressure using the current design. The logic in `AsyncCall.Post` today is "either enqueue, or drop". `Post` is synchronous, so the logic could be changed to "enqueue or wait" without changing the caller code.

My biggest concern though is I really do not have a good alternative in mind for the current implementation of the BroadcastProcessor. Assuming I have a `sinks` collection and a new `telemetry` object these are the alternatives I have considered so far

1. Simple iteration
```csharp
foreach(var sink in sinks) {
  try {
    sink.Process(telemetry); 
  } catch (Exception e) {
    // Process error
  }
}
```
Problem: sinks invoked one at a time, making the pipeline slower than it could be

2. Parallel iteration
```csharp
Parallel.For(0, sinks.Count, (sinkId) => {  
  try {
    sink.Process(telemetry); 
  } catch (Exception e) {
    // Process error
  }
});
```
Problem: I am really worried about the cost of setting up the Parallel.For machinery for every single piece of telemetry, esp. since the number of sinks will usually be small.

3. The current design, which is essentially a queue-per-sink, individually populated

@SergeyKanzhelev can you think of anything else?